phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -3707,15 +3709,20 @@ protected PhoenixConnection upgradeSystemCatalogIfRequired(PhoenixConnection met
             metaConnection = addColumnsIfNotExists(
                 metaConnection,
                 PhoenixDatabaseMetaData.SYSTEM_CATALOG,
-                MIN_SYSTEM_TABLE_TIMESTAMP_4_16_0 - 1,
+                MIN_SYSTEM_TABLE_TIMESTAMP_4_16_0 - 2,
```

Comment:
I'm a little unclear on these changes. Shouldn't all these 4.16-specific columns be added at the 4.16 ts rather than `4.16 ts - something`?

Reply:
@ChinmaySKulkarni  The upgrade code requires that each new column has a distinct timestamp (for some reason)